tiledb/sm/query/query.cc

```diff
@@ -947,6 +948,14 @@ Status Query::set_layout(Layout layout) {
   return Status::Ok();
 }
 
+Status Query::set_condition(const QueryCondition& condition) {
```

Comment:
We are going to need getters also for serialization work.

Reply:
Added a getter to the `Reader`, which can be fetched from the query with `query->reader()->condition()`.